run-build-functions.sh

```diff
@@ -98,19 +99,32 @@ run_yarn() {
     export PATH=$NETLIFY_CACHE_DIR/yarn/bin:$PATH
   fi
 
+  yarn_version=$(yarn --version)
 
-  echo "Installing NPM modules using Yarn version $(yarn --version)"
+
+  echo "Installing NPM modules using Yarn version $yarn_version"
   run_npm_set_temp
 
-  # Remove the cache-folder flag if the user set any.
+  local yarn_local="$YARN_FLAGS"
+
   # We want to control where to put the cache
   # to be able to store it internally after the build.
-  local yarn_local="${YARN_FLAGS/--cache-folder * /}"
-  # The previous pattern doesn't match the end of the string.
-  # This removes the flag from the end of the string.
-  yarn_local="${yarn_local%--cache-folder *}"
+  if [[ "${yarn_version%%.*}" -gt "1" ]]
+  then
+    # The cache-folder flag is deprecated in Yarn 2
+    # Override the environment variable if the user set any.
+    export YARN_GLOBAL_FOLDER="$NETLIFY_BUILD_BASE/.yarn_cache"
+    export YARN_ENABLE_GLOBAL_CACHE=1
```

Comment:
I'm not a fan of using environment variables for this, because it'll break people using zero-installs (when the cache is inside the repository), overriding the config they explicitly requested (I know it's what the code already did, overriding `--cache-folder` etc, but that really strikes me as a bad idea).

I think a more gentle approach would be to create a `.yarnrc.yml` file in the home directory that would contain the "default options" you want to have. This way, users would still be able to opt-out of the default workflow to use their own.

Reply:
@arcanis Thank you for the review! I've removed `export YARN_ENABLE_GLOBAL_CACHE=1`. The user now can opt-in to use global cache by setting `YARN_ENABLE_GLOBAL_CACHE` in Netlify settings.

I guess it'd be for Netlify to decide whether to allow users to override the global folder, so I'll leave it for now.

Reply:
Chiming in here... :)

I've also attempted to use zero-installs on Netlify and ran into issues as the install command [always seems to run](https://community.netlify.com/t/skip-yarn-install-when-using-yarn-2-with-zero-install/19423/4). Would it be possible to amend these changes to allow skipping of the install step completely? Or did I just miss how to do that 🙃 ?

Reply:
@kara-todd AFAIK, shoudn't `yarn install` only take few seconds with zero-installs? You can set `YARN_ENABLE_NETWORK=0` to disallow making network requests.

Reply:
@ylemkimon when I was running it... it still seemed to run relink step but I didn't try this with `YARN_ENABLE_NETWORK=0`. It didn't take terribly long as I recall, but I use this in conjunction with Netlify CMS which has to wait on builds for content reviews... So every second is ends up being precious. :)

I know that part of the problem was from Netlify not correctly detecting the cache directory as you are trying to fix here. Still a true `zero install` where I only run my build command would be preferable to me if it's not a huge lift.

Reply:
@kara-todd I have a Netlify repo with zero-install and it takes less than two seconds running `yarn install`. If it takes more than a few seconds, the repo may not have been set up properly. A workaround for now would be move the whole project into a subdirectory, not having `package.json` and `yarn.lock` at the root, and `cd` in the build command.